crawler.js

```diff
@@ -433,6 +437,20 @@ export class Crawler {
     }
   }
 
+  async shouldRunBehavior(frame) {
+    if (!frame.parentFrame()) {
+      return true;
+    }
+
+    const url = frame.url();
+
+    if (url === "about:blank") {
+      return false;
+    }
+
+    return await this.adBlockRules.shouldBlock(null, url);
```

Comment:
I think this will return `null` or `"advertisement"` rather than a boolean - does that work as expected in the filter?